File: week12/practice12_skeleton/search_skeleton.py
# Practice 12. Search
import sys
BINARY_SEARCH = 'B'
PERFECT_SQUARE = 'P'

if __name__ == "__main__":
  if len(sys.argv) != 3:
    raise Exception("Correct usage: [program] [input] [output]")
  with open(sys.argv[1], 'r') as inFile:
    lines = inFile.readlines()
  with open(sys.argv[2], 'w') as outFile:
    i = 0
    while i < len(lines):
      line = lines[i]
      words = line.split()
      op = words[0]
      if op == BINARY_SEARCH:
        if len(words) != 3:
          raise Exception("BINARY_SEARCH: invalid input")
        size, x = int(words[1]), int(words[2])
        # TODO
      elif op == PERFECT_SQUARE:
        if len(words) != 2:
          raise Exception("PERFECT_SQUARE: invalid input")
        x = int(words[1])
        # TODO
      else:
        raise Exception("Undefined operator")
      i += 1

        

File: week12/practice12_skeleton/search_skeleton.cpp
// Practice 12. Search
#include <cstdlib>
#include <cstring>
#include <fstream>
#include <iostream>
#include <sstream>
#include <tuple>
#include <vector>
#include <utility>
using namespace std;
const char BINARY_SEARCH = 'B';
const char PERFECT_SQUARE = 'P';

int main(int argc, char* argv[]) {
  if (argc != 3) {
    cerr<<"Correct usage: [program] [input] [output]"<<endl;
    exit(1);
  }
  ifstream inFile(argv[1]);
  ofstream outFile(argv[2]);
  string line;
  while (getline(inFile, line))
  {
    char op = line[0];
    istringstream iss(line.substr(2));
    vector<int> arr;
    int size, x, index;
    switch(op)
    {
      case BINARY_SEARCH:
        // TODO
        break;
      case PERFECT_SQUARE:
        if (!(iss >> x)) {
          cerr<<"Error: invalid input"<<endl;
          exit(1);
        }
        // TODO
        break;
      default:
        cerr<<"Undefined operator"<<endl;
        exit(1);
    }
  }
  outFile.close();
  inFile.close();
}
